app/code/Magento/Customer/Block/Account/AuthenticationPopup.php

```diff
@@ -48,6 +59,16 @@ public function __construct(
      */
     public function getJsLayout()
     {
+        // Check if captcha is not enabled and user is not logged in
+        if (!$this->_scopeConfig->getValue(
+            Form::XML_PATH_CUSTOMER_CAPTCHA_ENABLED,
+            ScopeInterface::SCOPE_STORE
+        ) && !$this->isLoggedIn()) {
```

Comment:
I don't think it can be related to whether the customer logged in or not. I would expect that this entire block won't be rendered then, but that's a separate issue.